### PR TITLE
Adding output binding for commercetools

### PR DESCRIPTION
**Pull request by harrykimpel**
# Description

Adding output binding for commercetools.

## Issue reference

n/a

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: https://github.com/dapr/docs/pull/1876

### CODE REVIEW

**Comment by harrykimpel** [comment]
@CodeMonkeyLeet can you pls. have a look at this again ... I would like to get this either continued or closed. Thx!

File: bindings/commercetools/commercetools.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	}
+
+	return res, nil
+}
+
+// getCommercetoolsMetadata returns new commercetools metadata.
+func (ct *Binding) getCommercetoolsMetadata(metadata bindings.Metadata) (*commercetoolsMetadata, error) {
+	meta := commercetoolsMetadata{}
+
+	val, ok := metadata.Properties["region"]
```
Comment (CodeMonkeyLeet) [inline bindings/commercetools/commercetools.go:135]: Suggestion: it's a fairly common pattern in the other components to collapse the two error cases into one, so feel free to do so:
```go
if val, ok := metadata.Properties["region"]; ok && val != "" {
	meta.region= val
} else {
	return nil, errors.New("commercetools error: missing `region` configuration")
}
```

**Review by CodeMonkeyLeet — COMMENTED** [review]
Sorry, this fell off my radar! The changes look good in general, the go.sum conflict just needs to be resolved and we should be good to go after that.

**Comment by harrykimpel** [comment]
> Sorry, this fell off my radar! The changes look good in general, the go.sum conflict just needs to be resolved and we should be good to go after that.

I will look into the error case suggestion.

Regarding the go.sum conflict, is this something I should be able to fix?

File: bindings/commercetools/commercetools.go (hunk 1 above)
Comment (harrykimpel) [reply bindings/commercetools/commercetools.go]: Followed your suggestion and collapsed all these separate error cases into one

**Comment by CodeMonkeyLeet** [comment]
> Regarding the go.sum conflict, is this something I should be able to fix?

Yes, you just need to make sure your branch is synced to the latest dapr master and run `go mod tidy`, which should resolve the go.mod changes for you.

**Comment by harrykimpel** [comment]
> > Regarding the go.sum conflict, is this something I should be able to fix?
> 
> Yes, you just need to make sure your branch is synced to the latest dapr master and run `go mod tidy`, which should resolve the go.mod changes for you.

I hope I was able to fix the go.sum issue

**Comment by CodeMonkeyLeet** [comment]
The comment change tripped the linter and the specific error can be seen in the Files changed view, could you fix that up?

From the previous run, this PR will also cause the certification tests to flag the tests' go.mod files as being out of sync, so `go mod tidy` will pretty much need to be run on each test under `tests/certification` as well.

@artursouza this will likely affect any PR in the queue before the cert test changes went in, and the repo hasn't really set appropriate expectations since then that contributors will also need to update the go.mod files of the cert tests. I've pushed https://github.com/dapr/components-contrib/pull/1308 to make that less of a chore, but should this PR wait on that?

**Comment by harrykimpel** [comment]
> The comment change tripped the linter and the specific error can be seen in the Files changed view, could you fix that up?
> 
> From the previous run, this PR will also cause the certification tests to flag the tests' go.mod files as being out of sync, so `go mod tidy` will pretty much need to be run on each test under `tests/certification` as well.

I have fixed the linter issue and also ran `go mod tidy`on the tests as well.

**Review by CodeMonkeyLeet — APPROVED** [review]
LGTM! Thanks for all the effort you put into the contribution, it's much appreciated!

**Comment by harrykimpel** [comment]
> LGTM! Thanks for all the effort you put into the contribution, it's much appreciated!

Thank you, my pleasure! It's much fun working on it!

File: bindings/commercetools/commercetools.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+
+	"github.com/labd/commercetools-go-sdk/commercetools"
```
Comment (artursouza) [inline bindings/commercetools/commercetools.go:16]: Just confirming that licensing is MIT, LGTM.

**Review by berndverst — DISMISSED** [review]

**Review by beiwei30 — APPROVED** [review]

**Comment by yaron2** [comment]
@harrykimpel there's a test failure with a generic error that probably hides the correct error. Can you run `go test -v ./...` and see what fails?

**Comment by harrykimpel** [comment]
> 

@yaron2 I did ran `go test -v ./...` just now, but did not notice any generic failed tests

**Comment by harrykimpel** [comment]
> > 
> 
> @yaron2 I did ran `go test -v ./...` just now, but did not notice any generic failed tests

okay, so on my Mac (arm-based) I did not see any tests fails, but on Windows (AMD, Ubuntu, WSL) I am getting this error message:

`build github.com/Azure/azure-sdk-for-go/sdk/azcore: cannot load github.com/Azure/azure-sdk-for-go/sdk/azcore: no Go source files`

File: bindings/commercetools/commercetools.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
```
Comment (dmitsh) [inline bindings/commercetools/commercetools.go:4]: After joining CNCF we have shifted to Apache License Version 2.0. Please update.

**Review by dmitsh — CHANGES_REQUESTED** [review]

**Comment by daixiang0** [comment]
@harrykimpel please fix DCO issue as well.

**Comment by berndverst** [comment]
@harrykimpel can you please sign your commits?

Also, please run `make modtidy-all` and commit those changes.

**Pull request closed without merging**